Source/EventFlow/Sagas/AggregateSagas/AggregateSaga.cs

```diff
@@ -68,11 +69,20 @@ public virtual async Task PublishAsync(ICommandBus commandBus, CancellationToken
         {
             var commandsToPublish = _unpublishedCommands.ToList();
             _unpublishedCommands.Clear();
+            var fails = new List<IExecutionResult>();
 
             foreach (var unpublishedCommand in commandsToPublish)
             {
-                await unpublishedCommand(commandBus, cancellationToken).ConfigureAwait(false);
+                var executionResult = await unpublishedCommand(commandBus, cancellationToken).ConfigureAwait(false);
+                if (!executionResult.IsSuccess)
+                    fails.Add(executionResult);
             }
+
+            if (0 < fails.Count)
+                if (1 == fails.Count)
+                    throw DomainError.With(fails[0].ToString());
```

Comment:
I would really like to see the failed execution results added to the exception. That way developers get more information on what went wrong and don't need to rely on `.ToString()` on their executions results.